src/richie/plugins/lti_consumer/models.py

```diff
@@ -0,0 +1,122 @@
+"""
+LTI consumer plugin models
+"""
+from urllib.parse import unquote
+
+from django.conf import settings
+from django.db import models
+from django.utils.functional import lazy
+from django.utils.translation import gettext_lazy as _
+
+import exrex
+from cms.models.pluginmodel import CMSPlugin
+from cms.utils import get_current_site
+from oauthlib import oauth1
+
+from .defaults import INSTRUCTOR, PREDEFINED_LTI_PROVIDERS, STUDENT
+
+
+class LtiConsumer(CMSPlugin):
+    """
+    LTI consumer plugin model.
+    """
+
+    url = models.URLField(
+        verbose_name=_("LTI url"),
+        help_text=_(
+            "For a predefined provider, leave this field empty for uploading new content"
+        ),
+        null=False,
+        blank=True,
+    )
+    lti_provider_id = models.CharField(
+        verbose_name=_("Predefined LTI provider"),
+        max_length=50,
+        null=True,
+        blank=True,
+        choices=lazy(lambda: PREDEFINED_LTI_PROVIDERS, tuple)(),
+        help_text=_("Please choose a predefined provider, or fill fields below"),
+    )
+    oauth_consumer_key = models.CharField(max_length=50, null=True, blank=True)
+    shared_secret = models.CharField(max_length=50, null=True, blank=True)
+
+    # pylint: disable=signature-differs
+    def save(self, *args, **kwargs):
+        """
+        Initialize attributes :
+        - context_id from current site domain
+        - url if predefined provider is used
+        """
+        if not self.id:
+            if self.lti_provider and not self.url:
+                self.url = "{:s}".format(
+                    exrex.getone(self.lti_provider.get("base_url")),
```

Comment:
This will fail if base_url is not set. Add a test?
```suggestion
                    exrex.getone(self.lti_provider.get("base_url", "")),
```